apis/python/src/tiledbsc/soma.py

```diff
@@ -387,22 +405,84 @@ def write_annotation_matrices(self, annotation_matrices, name: str):
         return subgroup_uri
 
     # ----------------------------------------------------------------
-    def __create_coo_array(self, uri, dim_names, attr_name):
+    def __create_annot_matrix(self, uri: str, mat, mat_name: str, dim_name: str, attr_names):
+        """
+        Create a TileDB 1D sparse array with string dimension and multiple attributes
+
+        :param uri: URI of the array to be created
+        :param mat: e.g. anndata.obsm['X_pca'] -- nominally a numpy.ndarray
+        :param mat_name: e.g. 'X_pca' in anndata.obsm['X_pca']
+        :param dim_name: name of the TileDB array dimension -- nominally 'obs_id' or 'var_id'
+        :param attr_names: column names for the dataframe
+        """
+        assert isinstance(uri, str)
+        assert isinstance(mat_name, str)
+        assert isinstance(dim_name, str)
+
+        # Nominally 'obs_id' or 'var_id'
+        dom = tiledb.Domain(
```

Comment:
I'll play around a bit with the compressors on this based on what the different datatype looks like